worksheet_D/oxo.py

```diff
@@ -1,26 +1,50 @@
 class OxoBoard:
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.gameboard = [0, 0, 0, 0, 0, 0, 0, 0, 0]
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        square = x + y * 3
+        return self.gameboard[square]
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        square = x + y * 3
+        if self.gameboard[square] == 0:
+            self.gameboard[square] = mark
+            return True
+        else:
+            return False
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        for squares in xrange(9):
+            if self.gameboard[squares] == 0:
+                return False
+
+        return True
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+        for x in xrange(3):
+
+            # Checks for horizontal and vertical lines of 3
+            if self.gameboard[x] == self.gameboard[x+3] and self.gameboard[x+3] == self.gameboard[x+6]:
```

Comment:
This needs a comment to explain the maths. Or maybe you could use `self.get_square` to make it more self-explanatory.

Note also that Python lets you shorten this:
``` if a == b and b == c: ```
to this:
``` if a == b == c: ```